src/algorithms/distance_transform.rs

```diff
@@ -29,67 +31,168 @@ pub fn chebyshev_distance_transform_from_terrain(
 /// This allows for calculating the distance transform from an arbitrary set of
 /// positions. Other position values in the initial Cost Matrix should be
 /// initialized to 255 (u8::MAX) to ensure the calculations work correctly.
-pub fn chebyshev_distance_transform_from_cost_matrix(
-    initial_cm: LocalCostMatrix,
-) -> LocalCostMatrix {
-    // Copy the initial cost matrix into the output cost matrix
-    let mut cm = initial_cm.clone();
```

Comment:
We actually never needed this clone, since we get the cost-matrix by value; the old owner can't do anything with it.